test/extended/authentication/oidc.go

```diff
@@ -0,0 +1,548 @@
+
+							copiedOC := *oc
+							tokenOC := copiedOC.WithToken(keycloakCli.AccessToken())
+							ssr, err = tokenOC.KubeClient().AuthenticationV1().SelfSubjectReviews().Create(ctx, &authnv1.SelfSubjectReview{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: fmt.Sprintf("%s-info", username),
+								},
+							}, metav1.CreateOptions{})
+							gomega.Expect(err).NotTo(o.HaveOccurred(), "should be able to create a SelfSubjectReview")
+						}).WithTimeout(5 * time.Minute).WithPolling(10 * time.Second).Should(o.Succeed())
+					})
+
+					g.It("should map UID correctly", func() {
+						o.Expect(ssr.UID).NotTo(o.Equal(strings.ToUpper(username)))
+					})
+
+					g.It("should map Extra correctly", func() {
+						o.Expect(ssr.Status.UserInfo.Extra).To(o.HaveKey("payload/test"))
+						o.Expect(ssr.Status.UserInfo.Extra["payload/test"]).To(o.HaveLen(1))
+						o.Expect(ssr.Status.UserInfo.Extra["payload/test"][0]).To(o.Equal(fmt.Sprintf("%s@payload.openshift.ioextra", username)))
+					})
+				})
+			})
+
+			g.Describe("[Skipped] with invalid specified UID or Extra claim mappings", func() {
+				g.It("should reject admission when UID claim expression is not compilable CEL", func() {
+					g.Skip("functionality not yet implemented")
+					/*
+						_, _, err := configureOIDCAuthentication(ctx, oc, func(o *configv1.OIDCProvider) {
+							o.ClaimMappings.UID = &configv1.TokenClaimOrExpressionMapping{
+								Expression: "!@&*#^",
+							}
+						})
+						o.Expect(err).To(o.HaveOccurred(), "should encounter an error configuring OIDC authentication")
+					*/
+				})
+
+				g.It("should reject admission when Extra claim expression is not compilable CEL", func() {
+					g.Skip("functionality not yet implemented")
+					/*
+						_, _, err := configureOIDCAuthentication(ctx, oc, func(o *configv1.OIDCProvider) {
+							o.ClaimMappings.Extra = []configv1.ExtraMapping{
+								{
+									Key:             "payload/test",
+									ValueExpression: "!@*&#^!@(*&^",
+								},
+							}
+						})
+						o.Expect(err).To(o.HaveOccurred(), "should encounter an error configuring OIDC authentication")
+					*/
+				})
+			})
+		})
+	})
+
+	g.AfterAll(func() {
+		err := removeResources(ctx, cleanups...)
+		o.Expect(err).NotTo(o.HaveOccurred(), "should not encounter an error cleaning up keycloak resources")
+
+		err = resetAuthentication(ctx, oc, originalAuth)
+		o.Expect(err).NotTo(o.HaveOccurred(), "should not encounter an error reverting authentication to original state")
+
+		waitForRollout(ctx, oc)
+	})
+})
+
+type removalFunc func(context.Context) error
+
+func removeResources(ctx context.Context, removalFuncs ...removalFunc) error {
+	errs := []error{}
+
+	for _, removal := range removalFuncs {
+		if removal == nil {
+			continue
+		}
+		err := removal(ctx)
+		if err != nil && !apierrors.IsNotFound(err) {
+			errs = append(errs, err)
+		}
+	}
+
+	return errors.Join(errs...)
```

Comment:
Nit:

[`errors.NewAggregate`](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/errors/errors.go) is very popular and widely used around our codebases. Otherwise, there seem to be some nice filtering capabilities as well:

```go
errors.FilterOut(errors.NewAggregate(errs), apierrors.IsNotFound)
```